modules/auxiliary/gather/dns_records.rb

```diff
@@ -0,0 +1,273 @@
+
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'net/dns/resolver'
+require 'rex'
+
+class Metasploit3 < Msf::Auxiliary
+  include Msf::Auxiliary::Report
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'         => 'Collect DNS Records Information',
+      'Description'  => %q{
+        This module can collect dns records (a, aaaa, mx, ns and so on).
+        Dns query and dns bruteforce will be used.
+      },
+      'Author'       => [ 'Nixawk' ],
+      'License'      => MSF_LICENSE
+    ))
+
+    register_options(
+      [
+        OptString.new('DOMAIN', [ true, "The target domain name" ]),
+        OptAddress.new('NS', [ false, "Specify nameserver for dns queries" ]),
+        OptPath.new('WORDLIST', [ false, "Wordlist file for subdomain bruteforce", ""]),
+        OptInt.new('TIMEOUT', [ false, "Dns response timeout", "4"]),
+        OptInt.new('THREADS', [ false, "Number of threads for subdomain bruteforce", 1]),
+        OptInt.new('RETRIES', [ false, "Times to retry (nameservers give us no response)", 2]),
+        OptInt.new('RETRY_INTERVAL', [ false, "Times to wait between first and second try)", 2]),
+      ], self.class)
+
+  end
+
+  def run
+    domain = datastore['DOMAIN'].to_s
+    nameserver = datastore['NS'].to_s
+    wordlist = datastore['WORDLIST'].to_s
+    threads = datastore['THREADS'].to_i
+    threads = 1 if threads == 0
+
+    @dns = Net::DNS::Resolver.new()   # initialize it with args
+
+    retry_interval = datastore['RETRY_INTERVAL'].to_i
+    @dns.retry_interval = 1 if retry_interval == 0
+
+    udp_timeout = datastore['TIMEOUT'].to_i
+    @dns.udp_timeout = 4 if udp_timeout == 0
+
+    retry_number = datastore['RETRIES'].to_i
+    @dns.retry_number = 2 if retry_number == 0
+
+    dns_ns_set(nameserver) unless nameserver.blank?
+    dns_wildcard(domain)
+    record = get_ns(domain, "NS")
+
+    get_a(domain, "A")
+    get_a(domain, "AAAA")
+    get_cname(domain, "CNAME")
+    get_mx(domain, "MX")
+    get_soa(domain, "SOA")
+    get_txt(domain, "TXT")
+
+    dns_bruteforce(domain, wordlist, threads) unless wordlist.blank?
+  end
+
+  def dns_ns_set(nameserver)
+    print_status("set dns nameserver : #{nameserver}")
+    @dns.nameserver = (nameserver.to_s)
+  end
+
+  def dns_query(domain, type="A")
+    print_status("enumerate [#{domain}] - #{type} record ")
+    @dns.query(domain, type)
+  end
+
+  def dns_wildcard(domain)
+    prefix = Rex::Text::rand_text_alpha(30) # random domain for dns wildcard check
+    result = get_a("#{prefix}.#{domain}")
+
+    unless result[:address].blank?
```

Comment:
Not wrong, but it [is not recommended](https://github.com/bbatsov/ruby-style-guide#no-else-with-unless) to use ```unless``` with ```else```.